### PR TITLE
New Dashboard: Fix "build a dashboard" when empty dash page feature is enabled

### LINKED ISSUE #66659: Panel edit plugin not found: undefined on a new dashboard screen
**What happened**:
During a usability testing, participant created a Prometheus data source, opened Data sources page (`/datasources`) and clicked on `Build a dashboard` button in the new Prometheus data source tile. On the New dashboard page, they saw an error – `Panel edit plugin not found: undefined`. I replicated it, and I am attaching a screen recording. 

image

**What you expected to happen**:
That they would see an empty dashboard. 

**How to reproduce it (as minimally and precisely as possible)**:
Open `/datasources` and click on `Build a dashboard` button for any of the default datasources. 

**Anything else we need to know?**:

**Environment**:
- Grafana version: Grafana v9.5.0-cloud.4.a016665c (57d4bf9f7e)

https://user-images.githubusercontent.com/4126171/232460490-f8bded49-99ed-454a-8faa-7ecce23864ca.mov

### PR DESCRIPTION
**Pull request by kaydelaney**
Fixes an issue where the "build a dashboard" button brought you to a broken dashboard when the new empty dashboard page feature is enabled.
Not sure what I've done here is the "best" way to get the datasource uid from A to B, but it seemed better than drilling a prop all the way down.

Closes #66659

### CODE REVIEW

File: public/app/features/dashboard/dashgrid/DashboardEmpty.tsx [hunk 1]
```diff
@@ -32,8 +37,9 @@ export const DashboardEmpty = ({ dashboard, canCreate }: Props) => {
             aria-label="Add new panel"
             onClick={() => {
               reportInteraction('Create new panel');
-              const id = onCreateNewPanel(dashboard);
+              const id = onCreateNewPanel(dashboard, initialDatasource);
```
Comment (polibb) [inline public/app/features/dashboard/dashgrid/DashboardEmpty.tsx:40]: we would need the same functionality here: https://github.com/grafana/grafana/blob/main/public/app/features/dashboard/components/AddPanelButton/AddPanelMenu.tsx#L30

File: public/app/features/dashboard/containers/NewDashboardWithDS.tsx [hunk 2]
```diff
@@ -17,19 +22,24 @@ export default function NewDashboardWithDS(props: GrafanaRouteComponentProps<{ d
...
+      const { dashboard } = newDashboard;
+      dashboard.panels[0] = {
+        ...dashboard.panels[0],
+        datasource: {
+          uid: ds.uid,
+          type: ds.type,
+        },
+      };
+
+      setDashboardToFetchFromLocalStorage(newDashboard);
```
Comment (polibb) [inline public/app/features/dashboard/containers/NewDashboardWithDS.tsx:36]: the emptyDashboardPage FF is enabled by default now, are you sure we don't need `setDashboardToFetchFromLocalStorage(newDashboard);` in the case where it is enabled too? I don't know why we have it in the first place, that's why I'm asking

File: public/app/features/dashboard/utils/dashboard.ts [hunk 3]
```diff
@@ -7,11 +7,12 @@ import store from 'app/core/store';
 import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
 import { calculateNewPanelGridPos } from 'app/features/dashboard/utils/panel';
 
-export function onCreateNewPanel(dashboard: DashboardModel): number | undefined {
+export function onCreateNewPanel(dashboard: DashboardModel, datasource?: string): number | undefined {
   const newPanel: Partial<PanelModel> = {
     type: 'timeseries',
     title: 'Panel Title',
     gridPos: calculateNewPanelGridPos(dashboard),
+    datasource: datasource ? { uid: datasource } : undefined,
```
Comment (polibb) [inline public/app/features/dashboard/utils/dashboard.ts:15]:
```suggestion
    datasource: datasource ? { uid: datasource } : null,
```

File: public/app/features/dashboard/containers/NewDashboardWithDS.tsx [hunk 4]
```diff
@@ -1,14 +1,19 @@
...
+import { useDispatch } from 'app/types';
 
 import { getNewDashboardModelData, setDashboardToFetchFromLocalStorage } from '../state/initDashboard';
+import { setInitialDatasource } from '../state/reducers';
 
 export default function NewDashboardWithDS(props: GrafanaRouteComponentProps<{ datasourceUid: string }>) {
   const [error, setError] = useState<string | null>(null);
   const { datasourceUid } = props.match.params;
+  const dispatch = useDispatch();
+  useLoadDataSource(datasourceUid);
```
Comment (polibb) [inline public/app/features/dashboard/containers/NewDashboardWithDS.tsx:16]: Why do we need this?

File: public/app/features/dashboard/containers/NewDashboardWithDS.tsx (hunk 2 above)
Comment (kaydelaney) [reply public/app/features/dashboard/containers/NewDashboardWithDS.tsx:36]: Pretty sure. In this case it (was) used to set the initial dashboard that includes the panel with the pre-set datasource, but that behavior is handled with setInitialDatasource now

File: public/app/features/dashboard/containers/NewDashboardWithDS.tsx (hunk 4 above)
Comment (kaydelaney) [reply public/app/features/dashboard/containers/NewDashboardWithDS.tsx:16]: Oops, we don't. Leftover from earlier attempt.

**Review by ivanortegaalba — APPROVED** [review]
All good from my side:
✅ The default DS is selected
✅ The bug is not there anymore
✅ When the DS doesn't exist, it doesn't break
✅ When adding a visualization without defaults, the flow works

**Pull request merged**